scripts/checkLicenses.ts

```diff
@@ -0,0 +1,124 @@
+ */
+function checkYear(
+    sourceFile: string,
+    [start, end]: string[],
+    callback: (err?: string) => void
+): void {
+    const currentYear = String(new Date().getFullYear());
+    exec(
+        [
+            // Added
+            `git --no-pager log --follow --diff-filter=A -n 1 --format=%ad --date=format:%Y -- ${sourceFile}`,
+
+            // Last modified
+            `git --no-pager log --follow --diff-filter=MRCA -n 1 --format=%ad --date=format:%Y -- ${sourceFile}`
+        ].join(" && "),
+        (error, stdOut, stdErr) => {
+            const copyrightStart = start;
+            const hasRange = end !== undefined;
+            const copyrightEnd = hasRange ? end : start;
+            const found = `${copyrightStart}${hasRange ? "-" + copyrightEnd : ""}`;
+            if (error) {
+                callback(error.message);
+            } else if (stdErr) {
+                callback(stdErr.toString());
+            } else if (stdOut.toString() === "") {
+                // The file is not in Git repository yet:
+                if (found !== currentYear) {
+                    callback(`${sourceFile} expected: ${currentYear}, found: ${found}`);
+                } else {
+                    callback();
+                }
+            } else if (!/^\d{4}\n\d{4}\n$/.test(stdOut.toString())) {
+                callback(`Can't determine first/last year of Git commit for file ${sourceFile}`);
+            } else {
+                const [firstCommit, lastCommit] = stdOut.toString().split("\n");
+                if (
+                    copyrightStart !== firstCommit ||
+                    copyrightEnd !== lastCommit ||
+                    (hasRange && copyrightStart === copyrightEnd)
+                ) {
+                    // Since a new commit is needed to fix it, the new commit must contain the current year:
+                    callback(
+                        `${sourceFile} expected: ${firstCommit}${
+                            firstCommit !== currentYear ? "-" + currentYear : ""
+                        }, found: ${found}`
+                    );
+                } else {
+                    callback();
+                }
+            }
+        }
+    );
+}
+
+/**
+ * Checks whether specified source files contain license matching the specified RegExp.
+ * @param sourceFiles The source files to check
+ * @param licenseRegEx RegExp to match the license.
+ *                      IMPORTANT, RegExp match should match the whole license and in the match groups returns the
+ *                      following:
+ *                      - match[1] - copyright start year
+ *                      - match[3] - copyright end year (if specified)
+ * @param callback The callback function to execute upon completion. Array of errors strings is passed back,
+ *                 in case not matching file is found.
+ */
+export function checkLicenses(
```

Comment:
Parameter fix needs to be documented ("if true, it tries to fix the year(s)")

Reply:
Done